include/seqan3/io/sequence_file/format_embl.hpp

```diff
@@ -195,6 +193,19 @@ class format_embl
         ++stream_it;
     }
 
+    /*!\brief Writes a number to the stream iterator using std::to_chars.
+     * \tparam stream_iterator_type The type of ostream iterator.
+     * \tparam number_type The type of number; must model seqan3::arithmetic.
+     * \param[in,out] stream_it The ostream iterator to write to.
+     * \param[in] num The number to write.
+     */
+    template <typename stream_iterator_type, arithmetic number_type>
+    auto write_number(stream_iterator_type & stream_it, number_type num)
+    {
+        auto res = std::to_chars(&arithmetic_buffer[0], &arithmetic_buffer[0] + sizeof(arithmetic_buffer), num);
+        stream_it.write_range(std::ranges::subrange<char *, char *>(&arithmetic_buffer[0], res.ptr));
+    }
```

Comment:
For historic reasons! Prior to this PR, we did not have our own stream_it type. I agree that now that we have it, `write_number()` and `write_eol` could become members of it. It could happen as part of this PR, but it could also happen separately.